DRAFT: AGIEval Benchmark | ASET - Arcadia Impact

bouromain opened the pull request:
## This PR contains:
- [x] New features
- [ ] Changes to dev-tools e.g. CI config / github tooling
- [ ] Docs
- [ ] Bug fixes
- [ ] Code refactor

### What is the current behavior? (You can also link to an open issue here)
This PR do not fix a current issue.

### What is the new behavior?
Implement the (AGIEval) benchmark for the english task

### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)

### Other information:

- the dataset used is the one in the [original repository of the benchmark]((https://github.com/ruixiangcui/AGIEval)).
- I currently only implemented the english tests. I started to implement the english ones and can provide it at a later time.
- Preliminary, evaluation are close to the ones reported in AGIEval repo. A few discrepancies are observed notably for the sat-math benchmark (only with GPT4o and GPT4o-mini, but not earlier models, thus suggesting data contamination)



### Preliminary evaluation (50 random samples)

#### zero-shot
 --  | paper |  paper |  inspect | inspect | inspect
-- | -- | -- | -- | -- | --
Test | gpt-3.5-turbo | gpt-4o | gpt-3.5-turbo | gpt-4o | gpt-4o-mini
aqua_rat | 56.7 | 56.7 | 40 | 94 | 86
logiqa_en | 38.6 | 55.6 | 30 | 56 | 54
lsat_ar | 27 | 31.3 | 30 | 34 | 28
lsat_lr | 53.3 | 84.9 | 52 | 88 | 70
lsat_rc | 60.6 | 90.3 | 70 | 94 | 82
sat_en | 70.4 | 93.2 | 78 | 96 | 92
sat_en_without_passage | 51.5 | 68.9 | 56 | 68 | 44
sat_math | 74.6 | 40.9 | 50 | 100 | 100
correlation with paper | -- | -- | 0.7314031866 | 0.5482362237 |  
correlation with paper (without sat_mat) |-- | -- |  0.8729996813 | 0.8333339019 |  


#### few-shots (5 examples)

 --  | paper |  paper |  inspect  
-- | -- | -- | --
Test | gpt-3.5-turbo | gpt-4o | gpt-3.5-turbo
aqua_rat | 35.4 | 75.9 | 32
logiqa_en | 45 | 59.3 | 32
lsat_ar | 30.4 | 20.8 | 16
lsat_lr | 59.6 | 88.8 | 58
lsat_rc | 66.9 | 92.2 | 66
sat_en | 51.4 | 93.7 | 82
sat_en_without_passage | 53.4 | 70.9 | 62
sat_math | 51.4 | 69.6 | 64
correlation with paper| -- | --| 0.7911706829

#### Cloze test (math)

 --  | paper |  paper |  inspect | inspect |
-- | -- | -- | -- | -- |
Test  | gpt-3.5-turbo | gpt-4o | gpt-3.5-turbo | gpt-4o |  
Zero shot | 30 | 67.4 | 32 | 70 |  
Few shots (5 samples) | 19 | 65 | 40 | 78 |  

jjallaire-aisi reviewed (commented):
Thanks for this! A reminder to also provide a README.md for the task. Also, a reminder to add this benchmark to the main README for benchmarks.

I am wondering about whether in its current form we need the task parameters (group and cloze) as group appears to be unused (for now) and you can select on cloze by just running the _cloze file. Perhaps though these are expansion joints that will come into play with further elaboration of the eval?

alexandraabbas commented on benchmarks/agieval/agieval_en.py:
```diff
@@ -0,0 +1,63 @@
+from inspect_ai import task
+
+
+## English Tasks
```
Would it make sense to combine all these tasks into one that takes `dataset_name` as an input argument?

alexandraabbas commented on benchmarks/agieval/agieval_cloze.py:
```diff
@@ -0,0 +1,11 @@
+from inspect_ai import task
+
+
+## Cloze Tasks
```
It might make sense to move this task to `agieval_en.py` since it uses the EN dataset.

bouromain replied:
I thought I would be easier to keep it in a separate file as the AGIEval_en in the paper consist in all MCQ tasks in english. It allows to run them more easily as they are in the same file. 

I am inclined to think it is more easy to keep it this way but happy to move it if needed.

bouromain commented:
> Thanks for this! A reminder to also provide a README.md for the task. Also, a reminder to add this benchmark to the main README for benchmarks.
> 
> I am wondering about whether in its current form we need the task parameters (group and cloze) as group appears to be unused (for now) and you can select on cloze by just running the _cloze file. Perhaps though these are expansion joints that will come into play with further elaboration of the eval?

I have added the READMEs it in the last commit. 

For the task parameters, you are right, I they only make sense if the tests in Chinese are also implemented.

jjallaire-aisi commented:
@bouromain Could you resolve the merge conflicts here? (I think hit's just the README table). 

Is this now at a place where you think we can merge or are their further revisions you want to make?

bouromain commented:
> @bouromain Could you resolve the merge conflicts here? (I think hit's just the README table).
> 
> Is this now at a place where you think we can merge or are their further revisions you want to make?

The conflict should be resolved now. 

In my view it is in a place were we can merge it yes. I think it will be more convenient if I add the Chinese test of this benchmark in an later PR.
